server/vulnerabilities/nvd/cve.go

```diff
@@ -632,7 +632,7 @@ func getMatchingVersionEndExcluding(ctx context.Context, cve string, hostSoftwar
 
 	// convert the host software version to semver for later comparison
 	formattedVersion := preprocessVersion(wfn.StripSlashes(hostSoftwareMeta.Version))
-	softwareVersion, err := semver.NewVersion(formattedVersion)
+	softwareVersion, err := fleet.VersionToSemverVersion(formattedVersion)
```

Comment:
This is failing because semver v3 considers`6.72.10-06` an invalid prerelease value.

V1 seems to be ok parsing it.
```
major: 6
minor: 72
patch: 10
pre: "06"
metadata: ""
original: "6.72.10-06"
```

Looking at the regex https://regex101.com/r/Ly7O1x/3/ this is indeed technically "not valid"

Reply:
https://github.com/fleetdm/fleet/blob/main/server/vulnerabilities/nvd/cve_test.go#L710-L719

Reply:
Added as part of https://github.com/fleetdm/fleet/pull/17683

Reply:
Changing this test to something like `6.72.10.1` and `6.72.10.2` would make this test pass. The `0\d+` pattern is what is considered "invalid"

Reply:
Going to skip this location as https://github.com/fleetdm/fleet/issues/24810 will most likely be removing semver usage here.